services/redis.yaml

```diff
@@ -0,0 +1,81 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: sd-redis
+  labels:
+    name: redis
+    app: sd-redis
+spec:
+  ports:
+  - port: 6379
+    name: master
+  clusterIP: None
+  selector:
+    name: redis
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: redis
+spec:
+  serviceName: "sd-redis"
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: sd-redis
+        name: redis
+        role: master
+      name: redis
+    spec:
+      initContainers:
+        - name: config-setter
+          image: busybox
+          env:
+            - name: REDIS_MASTER_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: redis-secret
+                  key: password
+          command: [
+            'sh', '-c',
+            'cp /redis-config-template/redis-master.conf /redis-conf/; sed -i "s/%master-password%/${REDIS_MASTER_PASSWORD}/" /redis-conf/redis-master.conf'
```

Comment:
@FenrirUnbound @minz1027 I stand corrected, `sed` is available inside redis container. But I still think it's a better pattern to extract out config related activities to an init container (so we don't have to make any assumptions on availability of unix commands inside main container.

Also with init container, our secret is not always exposed in actual container. Only init container will have access to k8s secret. I think this isolation of secrets is a better pattern in how secrets are exposed inside actual container.  (redis.conf stores it in plain text, but that is a different issue).